### PR TITLE
Run rustfmt

### PR DESCRIPTION
**Pull request by mcarton**
So I was going to integrate coverage tests in Travis, but [@Manishearth was to fast](https://github.com/Manishearth/rust-clippy/pull/1255#issuecomment-251158625).
So I though “Hey, what other tool could I fight against? Oh I know, [it's been a long time since we haven't run `rustfmt`](https://github.com/Manishearth/rust-clippy/pull/1249#issuecomment-250905291)!”.

:warning: I offer one Internet point to anyone who can explain to me how `rustfmt` decides what to do with `match` arms and another one for all our `|db| {}` lambdas.

### CODE REVIEW

**Comment by mcarton** [comment]
And of course #1246 is calling and there will be horrible merge conflicts I'm sure \o/

**Comment by mcarton** [comment]
No conflict whatsoever, where is the fun git?

File: clippy_lints/src/attrs.rs [hunk 1]
```diff
@@ -121,21 +121,23 @@ impl LateLintPass for AttrPass {
                                 }
                                 if let Some(mut sugg) = snippet_opt(cx, attr.span) {
                                     if sugg.len() > 1 {
-                                        span_lint_and_then(cx, USELESS_ATTRIBUTE, attr.span,
+                                        span_lint_and_then(cx,
+                                                           USELESS_ATTRIBUTE,
+                                                           attr.span,
                                                            "useless lint attribute",
                                                            |db| {
                                             sugg.insert(1, '!');
```
Comment (oli-obk) [inline clippy_lints/src/attrs.rs]: I actually like what rustfmt did with our closures. They look like this closure now.

File: clippy_lints/src/attrs.rs [hunk 2]
```diff
@@ -121,21 +121,23 @@ impl LateLintPass for AttrPass {
...
                                             db.span_suggestion(attr.span, "if you just forgot a `!`, use", sugg);
                                         });
                                     }
                                 }
-                            },
```
Comment (oli-obk) [inline clippy_lints/src/attrs.rs]: set `match_block_trailing_comma = true`

File: clippy_lints/src/attrs.rs (hunk 1 above)
Comment (oli-obk) [inline clippy_lints/src/attrs.rs]: try `closure_block_indent_threshold = -1` or `closure_block_indent_threshold = 99`

File: clippy_lints/src/attrs.rs (hunk 2 above)
Comment (mcarton) [inline clippy_lints/src/attrs.rs]: I don't really have a strong opinion on that one. The default looks fine.

Comment (oli-obk) [inline clippy_lints/src/attrs.rs]: I have a strong opinion :P I really like the fact that you can distinguish a match arm block closing bracket from other closing brackets, but I can do my own PR when I get around to it.

**Comment by oli-obk** [comment]
I reran rustfmt, there was a really degenerate case in one of our tests, but even I can't figure out how to format that properly, so I just slapped on a `rustfmt_skip`

**Comment by mcarton** [comment]
@oli-obk You realize this only runs rustfmt on `clippy` but not `clippy_lints` right?
The original PR (which by now had no chance to be rebased anyway) ran it on `clippy_lints` too.

**Comment by oli-obk** [comment]
whoops *fixing*

I was wondering why I got so few hits

**Comment by mcarton** [comment]
> I was wondering why I got so few hits

:smile:

**Pull request merged**